model-optimizer/extensions/front/EmbeddingBagFuse.py

```diff
@@ -0,0 +1,91 @@
+        return [ExpandDimsToUnsqueeze, AttributedGatherNormalizer]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('weights', dict(op='Const')),
+                ('concat_before', dict(op='Concat')),
+                ('gather_before1_1', dict(op='Gather')),
+                ('unsqueeze_before1_1', dict(op='Unsqueeze')),
+                ('gather_before2_1', dict(op='Gather')),
+                ('unsqueeze_before2_1', dict(op='Unsqueeze')),
+                ('slice1', dict(op='Slice')),
+                ('gather_after1', dict(op='Gather')),
+                ('reduce1', dict(op='ReduceSum')),
+                ('unsqueeze_after1', dict(op='Unsqueeze')),
+                ('concat_after', dict(op='Concat')),
+            ],
+            edges=[
+                ('concat_before', 'gather_before1_1'),
+                ('concat_before', 'gather_before2_1'),
+                ('gather_before1_1', 'unsqueeze_before1_1'),
+                ('gather_before2_1', 'unsqueeze_before2_1'),
+                ('unsqueeze_before1_1', 'slice1', {'out': 0, 'in': 1}),
+                ('unsqueeze_before2_1', 'slice1', {'out': 0, 'in': 2}),
+                ('weights', 'gather_after1', {'out': 0, 'in': 0}),
+                ('slice1', 'gather_after1', {'out': 0, 'in': 1}),
+                ('gather_after1', 'reduce1'),
+                ('reduce1', 'unsqueeze_after1'),
+                ('unsqueeze_after1', 'concat_after'),
+            ])
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        concat_before = match['concat_before']
+        gather_after1 = match['gather_after1']
+        slice1 = match['slice1']
+        concat_after = match['concat_after']
+        weights_node = gather_after1.in_port(0).get_source().node
+        gather_after_axis = gather_after1.in_port(2).get_source().node.soft_get('value')
+        for dst_port in weights_node.out_port(0).get_destinations():
+            node = dst_port.node
+            if node.op == 'Gather':
+                # validate that all Gathers have same axis
+                if node.in_port(2).get_source().node.soft_get('value') != gather_after_axis:
+                    return
+                dst_port.disconnect()
+        indices_node = slice1.in_port(0).get_source().node
+        slice_axis = slice1.in_port(3).get_source().node.soft_get('value')
+        for dst_port in indices_node.out_port(0).get_destinations():
+            node = dst_port.node
+            if node.op == 'Slice':
+                # validate that all Slices have same axis
+                if node.in_port(3).get_source().node.soft_get('value') != slice_axis:
+                    return
+                dst_port.disconnect()
+        emb_bag = EmbeddingBagOffsetsSum(graph, {}).create_node()
+        weights_node.out_port(0).connect(emb_bag.in_port(0))
+        indices_node.out_port(0).connect(emb_bag.in_port(1))
+        concat_before.in_port(0).get_connection().set_destination(emb_bag.in_port(2))
```

Comment:
Should be changed to `concat_before.in_port(0).get_source().connect(emb_bag.in_port(2))`. The reason for this is that some nodes may use internal operations of the pattern, for example, output of `gather_before_1_1` and may be connected with some "Result" operation. If you disconnect input of `concat_before` operation then these nodes will not have enough input data and this will result in partial shape failure. But if you use `connect` method then since you disconnected output of `concat_after` to EmbeddingBag the unneeded sub-graph will be cleaned up later but possible other users of `gather_before_1_1` will stay.

Many our transformations suffer from similar bug. Lets not introduce more.